File: public-api/src/main/java/com/shopleech/publicapi/bll/service/model/IShopService.java
package com.shopleech.publicapi.bll.service.model;

import com.shopleech.publicapi.domain.Shop;

import java.util.List;

/**
 * @author Ahto Jalak
 * @since 05.04.2023
 */
public interface IShopService {
    List<Shop> getAll();

    Shop get(Integer id) throws Exception;

    Shop add(Shop shop) throws Exception;

    Shop update(Integer id, Shop shop) throws Exception;

    Integer remove(Integer id) throws Exception;

    List<Shop> search(Shop entity);
}

File: price-comparison-client1/src/bll/service/model/IShopService.ts
import type { IShop } from '@/dal/domain/IShop'
import type { IServiceResult } from '@/bll/service/model/IServiceResult'

/**
 * @author Ahto Jalak
 * @since 12.04.2023
 */
export interface IShopService {
    findByName (searchName: string): Promise<IServiceResult<IShop[]>>

    add (shopInfo: IShop): Promise<IServiceResult<IShop>>

    getAll (): Promise<IServiceResult<IShop[]>>
}
